### PR TITLE
Wiz v1.1.0 - take 2

### PR DESCRIPTION
**Pull request by solalraveh**
~~DRAFT. IN PROGRESS. PANW team, please don't review yet, we'll update via CS and Slack once ready.
We're PR-ing it, so the unit testing report os generated again (previous one expired). We're now working on it.~~

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

#### Commands
- Added `wiz-issue-in-progress` command. See docs for usage.
- Added `wiz-reopen-issue` command. See docs for usage.
- Added `wiz-reject-issue` command. See docs for usage.
- Added `wiz-set-issue-note` command. See docs for usage.
- Added `wiz-clear-issue-note` command. See docs for usage.
- Added `wiz-get-issue-evidence` command. See docs for usage.
- Added `wiz-rescan-machine-disk` command. See docs for usage.
- Added `wiz-set-issue-due-date` command. See docs for usage.
- Added `wiz-clear-issue-due-date` command. See docs for usage.
- Added `wiz-get-project-team` command. See docs for usage.

##### Wiz Mapper
- Add new Incident Field "Wiz Resource Cloud ID" which has the Cloud Provider ID, whereas Wiz Resource ID has the Wiz unique ID.
- Add new Incident Field "Wiz Issue URL" with URL to Wiz Issue.
- Add new Incident Field "Wiz Projects". This value can be empty if no Projects apply.
- Add new Incident Field "Wiz Projects Owners". This value can be empty if no Projects apply or if no Owners were assigned.
- Add new Incident Field "Wiz Projects Security Champions". This value can be empty if no Projects apply or if no Security Champions were assigned.

##### Wiz Mapper Webhook
- Add new Incident Field "Wiz Resource Cloud ID" which has the Cloud Provider ID, whereas Wiz Resource ID has the Wiz unique ID.
- Changed the transformation of INFORMATIONAL incidents from 0 to 0.5
- Add new Incident Field "Wiz Issue URL" with URL to Wiz Issue.
- Add new Incident Field "Wiz Projects". This value can be empty if no Projects apply.
- Add new Incident Field "Wiz Projects Owners". This value can be empty if no Projects apply or if no Owners were assigned.
- Add new Incident Field "Wiz Projects Security Champions". This value can be empty if no Projects apply or if no Security Champions were assigned.

##### Wiz
- Changed Demisto info logging to error in several output fields.
- Better error handling for VM disk rescan if VM isn't found.
- Better JSON parsing handling.

#### Incident Types
- **Wiz Issue**
- Fix for choosing the right pre-built Wiz Layout.

#### Incident Fields
- **Wiz Issue URL**
- **Wiz Resource Cloud ID**
- **Wiz Resource ID**
- Add new Incident Field "Wiz Resource Cloud ID" which has the Cloud Provider ID, whereas Wiz Resource ID has the Wiz unique ID.

##### Wiz Layout
- Added "Wiz Resource Cloud ID" to the Wiz Layout.
- Added "Wiz Issue URL" to the Wiz Layout.
- Added "Wiz Projects" to the Wiz Layout. This value can be empty if no Projects apply.
- Added "Wiz Projects Owners" to the Wiz Layout. This value can be empty if no Projects apply or if no Owners were assigned.
- Added "Wiz Projects Security Champions" to the Wiz Layout. This value can be empty if no Projects apply or if no Security Champions were assigned.

##### New: Assess Wiz Issues
- Example basic Playbook to assess Wiz Issues
- Added "Assess_Wiz_Issues" Playbook to be automatically mapped to Wiz Incident types, but not run automatically.

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by solalraveh** [comment]
@kgal-pan @ariel-wiz FYI, take 2.

**Comment by solalraveh** [comment]
@kgal-pan should be ready to go.
image
@ariel-wiz 👑

**Comment by ariel-wiz** [comment]
Hi @kgal-pan - Any update on this PR?

File: Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_Project_Owners.json
Comment (kgal-pan) [inline Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_Project_Owners.json:20]:
```suggestion
    "associatedTypes": [
        "Wiz Issue"
    ],
    "associatedToAll": false,
```

File: Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_Projects_Security_Champions.json
Comment (kgal-pan) [inline Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_Projects_Security_Champions.json:20]:
```suggestion
    "associatedTypes": [
        "Wiz Issue"
    ],
    "associatedToAll": false,
```

File: Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_URL.json
Comment (kgal-pan) [inline Packs/Wiz/IncidentFields/incidentfield-Wiz_Issue_URL.json:20]:
```suggestion
    "associatedTypes": [
        "Wiz Issue"
    ],
    "associatedToAll": false,
```

File: Packs/Wiz/Integrations/Wiz/Wiz.py
Comment (israelpoli) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:609]: I do not think I need the second condition,
I'm wrong?

```suggestion
    if not issue_id or not reject_reason or not reject_note:
```

File: Packs/Wiz/Integrations/Wiz/Wiz.py
Comment (israelpoli) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:762]:
```suggestion
    if not issue_id or not note:
```

File: Packs/Wiz/Integrations/Wiz/Wiz.py
Comment (israelpoli) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:793]:
```suggestion
    if not issue_response_json['issue'].get('note'):
```

File: Packs/Wiz/Integrations/Wiz/Wiz.py
Comment (israelpoli) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:834]: Does it clear all the notes?
If so then:
```suggestion
    Clear All notes from a Wiz Issue
```

File: Packs/Wiz/Integrations/Wiz/Wiz.py
Comment (israelpoli) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:905]:
```suggestion
    if not issue_id or not due_at:
        demisto.error("The issue ID and due_at must be provided")
        return "The issue ID and due_at must be provided"

   
    # time has to be like 2022-01-19T00:00:00.000Z
    format = "%Y-%m-%d"
    try:
        datetime.strptime(due_at, format)
        demisto.info("This is the correct date string format.")
    except ValueError:
        demisto.error("This is the incorrect. It should be YYYY-MM-DD")
        return "The date format is the incorrect. It should be YYYY-MM-DD"
    due_at = due_at + 'T00:00:00.000Z'
```

File: Packs/Wiz/Integrations/Wiz/Wiz.py
Comment (israelpoli) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:1176]:
```suggestion
    if response.get('data', {}).get('graphSearch', {}).get('nodes') is None:
```

File: Packs/Wiz/Integrations/Wiz/Wiz.py
Comment (israelpoli) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:1236]:
```suggestion
    if vm_response.get('data', {}).get('graphSearch', {}).get('nodes', []) == []:
```

File: Packs/Wiz/Integrations/Wiz/Wiz.py
Comment (israelpoli) [inline Packs/Wiz/Integrations/Wiz/Wiz.py:1357]:
```suggestion
    project_response = response_json.get('data', {}).get('projects', {}).get('nodes')
```

**Review by israelpoli — CHANGES_REQUESTED** [review]
Hi @solalraveh...
Thank you for your contribution

Nice job
Please see the comments I have added

And again great job
**Thanks**

**Review by idovandijk — CHANGES_REQUESTED** [review]
Thank you @solalraveh for the contribution! I've reviewed the parts not related to actual code in the PR. Good job with this PR! I have some comments, please address them:
1. I see that there's a "Wiz Classifier" classifier, but I don't see any classification going on there. Could it be a local issue for me? Please double check on your end.
2. The playbook sets the due date for the issue, but I am confused on where the due date incident field is populated in the flow. I don't see a mapping for it:
image
3. Could you please give the playbook tasks meaningful names and descriptions?
4. The playbook is hardcoded to get all high severity issues from Wiz. Could you make the severity and the issue type configurable from the playbook inputs to make this playbook easier for others to use?
image
5. In the release-notes, could you please move the line that explains that the incident type is mapped to a playbook, to be under the incident type instead of under the playbook?
6. Please rename this playbook to meet our naming standard: "Wiz - Assess Issues".

Once done with the changes can you please give us some time slots where we could see a short demo of the flow? As we cannot run such incidents on our environment I would be happy to see the playbook running and how the incident layout data is displayed.

Thanks!

**Comment by solalraveh** [comment]
The playbook is only meant as an example of how to use the commands and not as a binding and mandatory way to deal with incidents, as each customer prefers their own way. This hence is very high level. 
I also see comments on code standards that I'll change only if needed while back from OOO. 
thank you..

**Comment by ariel-wiz** [comment]
Hi @israelpoli and @idovandijk - Please see the requested code fixes.
@idovandijk, regarding your first comment about Wiz Classifier - We’ve created this classifier as a preparation for potential future differentiation between Wiz issue types. This has been discussed and vetted by @kgal-pan in a previous release. The code infrastructure has been built around it, so we want to let it as it for the moment.

**Comment by israelpoli** [comment]
Hi @ariel-wiz 
The Run Validations is failure, you need change the name of the playbook's README file to the name of playbook's yml file, like this:

Packs/Wiz/Playbooks/Wiz_-_Assess_Issues_README.md  ->  `Packs/Wiz/Playbooks/Assess_Wiz_Issues_README.md`

**Comment by ariel-wiz** [comment]
@israelpoli - Done
Looking forward for merging the code

**Comment by ariel-wiz** [comment]
@israelpoli Can you please clarify the reason circleci check has failed?

**Comment by israelpoli** [comment]
> @israelpoli Can you please clarify the reason circleci check has failed?

Hi @ariel-wiz 
It seems it falls down to not having a README for the playbook
Please change the name of the README of the playbook to:

`Packs/Wiz/Playbooks/Assess_Wiz_Issues_README.md`

**Comment by israelpoli** [comment]
@ariel-wiz 

You have to pull from the master and push.
But right now we are waiting for @idovandijk  to do his review for everything that is not code
So wait with it.

**Comment by idovandijk** [comment]
> The playbook is only meant as an example of how to use the commands and not as a binding and mandatory way to deal with incidents, as each customer prefers their own way. This hence is very high level. I also see comments on code standards that I'll change only if needed while back from OOO. thank you..

I understand the point of the playbook being an example and the classifier being set as basis for future releases.
We still need to give the playbook and tasks meaningful names/descriptions, to make that example actually an example that would be easier to adopt. For instance - `wiz-issue-in-progress` doesn't tell the user whether it's a GET or a SET.

Please make these small changes and let's schedule a short demo just to see the flow of the hard work you've done here. Thanks for your patience as you waited for our reviews.

**Comment by kgal-pan** [comment]
@ariel-wiz - Can you please join the [DFIR community](https://start.paloaltonetworks.com/join-our-slack-community) so we can coordinate the demo?

**Comment by kgal-pan** [comment]
@ariel-wiz @solalraveh  @idovandijk - Thanks for joining the call earlier.

As we discussed, the [tableToMarkdown](https://xsoar.pan.dev/docs/integrations/code-conventions#tabletomarkdown) outputs for the commands will be added in the next version (`1.2.0`) as there's an urgency to get this PR approved and through.

Please make sure to update the fork from upstream to resolve [the validation issue](https://app.circleci.com/pipelines/github/demisto/content/180105/workflows/388964f9-d4d0-4535-bf81-3496cbd84f71/jobs/548887?invite=true#step-102-7).

**Comment by idovandijk** [comment]
@israelpoli feel free to approve if you don't need any other changes.

**Comment by ariel-wiz** [comment]
Hi @israelpoli 
What is the approval status?
As discussed we need to deliver this ASAP to release a new version that supports the new authentication method.

**Comment by israelpoli** [comment]
> Hi @israelpoli What is the approval status? As discussed we need to deliver this ASAP to release a new version that supports the new authentication method.

Hi @ariel-wiz 
we have a problem
We are trying to solve it
The PR will merge immediately when the problem is solved.

sorry for it.

**Comment by ariel-wiz** [comment]
Thanks for the update @israelpoli 
Is there an ETA to solve the issue?

**Comment by israelpoli** [comment]
> Thanks for the update @israelpoli Is there an ETA to solve the issue?

@ariel-wiz 
1. Please add in the Assess_Wiz_Issues.yml file `tests: - No tests` as there is in wiz.yml file.
2. pull from master and push.

I hope this solves our problem.
Thanks

**Comment by ariel-wiz** [comment]
@israelpoli I've added the requested changes.

**Comment by ariel-wiz** [comment]
@israelpoli I see that one check has failed. Is the issue on our side?

**Comment by israelpoli** [comment]
> @israelpoli I see that one check has failed. Is the issue on our side?

Sorry for the delay in replying, but it was a bug on our part,
which is now fixed

**Comment by ariel-wiz** [comment]
@israelpoli is there something that blocks us from merging the code now?

**Review by israelpoli — APPROVED** [review]
Thanks a lot

**Comment by ariel-wiz** [comment]
@israelpoli Thank you for merging!
When the changes will be live in XSOAR?

**Comment by kgal-pan** [comment]
> @israelpoli Thank you for merging! When the changes will be live in XSOAR?

@ariel-wiz  - Once https://github.com/demisto/content/pull/20475 is approved and merged, it will take around 12 hours to update.

**Pull request merged**
